### PR TITLE
Nest leg under boarding pass

### PR DESCRIPTION
**Pull request by kantord**

### CODE REVIEW

File: src/booking/Booking.js [hunk 1]
```diff
@@ -83,4 +83,5 @@ export type BookingAssets = {
   ticketUrl: ?string,
   invoiceUrl: ?string,
   boardingPasses: { [string]: string },
+  legs: Array<Leg>,
```
Comment (jaroslav-kubicek) [inline src/booking/Booking.js]: => `Leg[]`

File: src/booking/types/outputs/BoardingPass.js [hunk 2]
```diff
@@ -13,5 +14,10 @@ export default new GraphQLObjectType({
       type: GraphQLString,
       resolve: ({ boardingPassUrl }) => boardingPassUrl,
     },
+    leg: {
+      type: Leg,
+      resolve: ({ legs, flightNumber }) =>
+        legs.filter(leg => leg.id === flightNumber)[0],
```
Comment (jaroslav-kubicek) [inline src/booking/types/outputs/BoardingPass.js]: the case when an array can be empty after filtering should be handled.

Comment (kantord) [reply src/booking/types/outputs/BoardingPass.js]: `[][0]` returns `undefined`, which is exactly what we want so I think it is handled implicitly. I added a default value just to make is explicit

File: src/booking/types/outputs/BookingAssets.js [hunk 3]
```diff
@@ -26,13 +26,15 @@ export default new GraphQLObjectType({
...
         boardingPasses,
+        legs,
       }: BookingAssets): Array<{
         boardingPassUrl: mixed,
         flightNumber: string,
       }> =>
         Object.entries(boardingPasses).map(([key, value]) => ({
           boardingPassUrl: value,
           flightNumber: key,
+          legs: legs,
```
Comment (tbergq) [inline src/booking/types/outputs/BookingAssets.js]: `legs: legs,` -> `legs,`

File: src/booking/types/outputs/BoardingPass.js [hunk 4]
```diff
@@ -13,5 +14,10 @@ export default new GraphQLObjectType({
       type: GraphQLString,
       resolve: ({ boardingPassUrl }) => boardingPassUrl,
     },
+    leg: {
+      type: Leg,
+      resolve: ({ legs, flightNumber }) =>
+        legs.filter(leg => leg.id === flightNumber)[0] || null,
+    },
```
Comment (mrtnzlml) [inline src/booking/types/outputs/BoardingPass.js]: Everyone is now talking about how we should use the Glossary and we are thinking about how to migrate our GraphQL. It didn't go very well here... 🙄 😬 (https://kiwi.wiki/#/comms/glossary)

Comment (kantord) [reply src/booking/types/outputs/BoardingPass.js]: So we should rename this to `segment`?

Comment (mrtnzlml) [reply src/booking/types/outputs/BoardingPass.js]: We will have to do it somehow systematically later I guess...

**Pull request merged**